src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -2753,3 +2758,42 @@ extern "C" void SystemNative_GetDomainSocketSizes(int32_t* pathOffset, int32_t*
     *pathSize = sizeof(domainSocket.sun_path);
     *addressSize = sizeof(domainSocket);
 }
+
+extern "C" Error SystemNative_SendFile(intptr_t out_fd, intptr_t in_fd, int64_t offset, int64_t count, int64_t* sent)
+{
+    int outfd = ToFileDescriptor(out_fd);
+    int infd = ToFileDescriptor(in_fd);
+
+#if HAVE_SENDFILE_4
+    off_t offtOffset = static_cast<off_t>(offset);
+
+    ssize_t res;
+    while (CheckInterrupted(res = sendfile(outfd, infd, &offtOffset, static_cast<size_t>(count))));
+    if (res != -1)
+    {
+        *sent = res;
+        return PAL_SUCCESS;
+    }
+
+    *sent = 0;
+    return SystemNative_ConvertErrorPlatformToPal(errno);
+#elif HAVE_SENDFILE_6
+    off_t len = 0;
+    ssize_t res;
+    while (CheckInterrupted(res = sendfile(infd, outfd, static_cast<off_t>(offset), &len, nullptr, 0)));
+    if (res != -1)
+    {
+        *sent = len;
+        return PAL_SUCCESS;
+    }
+
+    *sent = 0;
+    return SystemNative_ConvertErrorPlatformToPal(errno);
+#else
+    (void)outfd;
+    (void)infd;
```

Comment:
I think you need to add:
```C++
(void)offset;
(void)count;
```
as otherwise you'll likely get warnings/errors about unused parameters.
```
Otherwise